Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchExpression.cs

```diff
@@ -49,18 +50,22 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public SwitchExpression Update (Expression switchValue, IEnumerable<SwitchCase> cases, Expression defaultBody)
     {
+      Contract.Requires(switchValue != null);
```

Comment:
Ok, it seems proper precondition should be like that:

`Contract.Requires(cases != null && cases.Any());`
`Contract.Requires(Contract.ForAll(cases, case => case != null));`